src/coreclr/src/jit/importer.cpp

```diff
@@ -4272,7 +4272,8 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
 
     if (mustExpand && (retNode == nullptr))
     {
-        NO_WAY("JIT must expand the intrinsic!");
+        assert(!"Unhandled must expand intrinsic, throwing PlatformNotSupportedException");
+        return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
```

Comment:
Previously the JIT would fail fast with `The JIT compiler encountered invalid IL code or an internal limitation.`

Now, it will assert in debug mode but will `throw PlatformNotSupportedException` at runtime.

Reply:
This was changed to not use `impUnsupportedHWIntrinsic` and instead use `gtNewMustThrowException` directly.